homeassistant/components/device_tracker/tplink.py

```diff
@@ -52,9 +52,12 @@ def get_scanner(hass, config):
             TplinkDeviceScanner, Tplink5DeviceScanner, Tplink4DeviceScanner,
             Tplink3DeviceScanner, Tplink2DeviceScanner, Tplink1DeviceScanner
     ]:
-        scanner = cls(config[DOMAIN])
-        if scanner.success_init:
-            return scanner
+        try:
```

Comment:
This try/catch should move into `TplinkDevice[*]Scanner._update_info()`, wrapping `self.tplink_client.get_connected_devices()`

Reply:
Only TplinkDeviceScanner uses `self.tplink_client.get_connected_devices()` the others use `requests.get` or `requests.session`  You want me to wrap each individual call in a try except?

Reply:
Since scanner use a flag `success_init` to indicate whether initialize success, the try/catch should be done in `scanner.__init__` if need.